### ISSUE TITLE
Workspace trust dialog accessibility

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#122537]
I see two issues with the workspace trust dialog and accessibility

1. The description is not read out. " VS Code provides advanced editing features that may automatically execute files in this folder. If you don't trust the authors of these files, we recommend to continue in restricted mode as the files may be malicious. See our docs to learn more.". So this part is completely unaccessible to screen reader users
2. It is not possible to tab onto "our docs" link

**Comment by isidorn** [comment]
Let's add a bug label so I verify this later in the milestone.

**Comment by lramos15** [comment]
@isidorn Just pinging you again for the verification 🙂

**Comment by isidorn** [comment]
1. is still not fixed. As soon as this dialog gets shown the content of it is not read out. Only the button is. I think this can be fixed by puting the focus initally on the whole dialog, the dialog would have the right aria-label that has the same value as the content. And from there user can tab navigate to the buttons
2. is fixed

Reopening. Though I think it is fine to ship this, but would be good if we fix this next milestone

**Comment by sbatten** [comment]
closed by https://github.com/microsoft/vscode/commit/bf013916b4f7155ec2c399d0f569d5ebf94799fa

**Comment by sbatten** [comment]
@isidorn based on guidance [here](https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html#:~:text=Role%2C%20Property%2C%20State%2C%20and%20Tabindex%20Attributes%20%20,Tells%20assistive%20technologies%20that%20the%20wi%20...%20), I've used a combination of tabindex=-1 and aria-describedby to achieve the desired result across platforms.

**Comment by isidorn** [comment]
@sbatten that sounds good. Let's close this so I verify this later today and then I can provide more feedback if needed :)

**Comment by isidorn** [comment]
I have tried it out and it seems good. Here's what NVDA reads when this dialog appears

"Do you trust the authors of the files in this folder?  dialog  Info Do you trust the authors of the files in this folder? Code - Insiders provides features that may automatically execute files in this folder. If you don't trust the authors of these files, we recommend to continue in restricted mode as the files may be malicious. See our docs to learn more. C:\Development\ee\Dictionaries"

So there seems to be repetition "Do you trust the authors of the files in this folder" is announced twice. Could we improve that?

As for the code change I see that you nicely used the `describedby` however the value seems to be somewhat complex
`aria-describedby="monaco-dialog-icon monaco-dialog-message-text monaco-dialog-message-detail monaco-dialog-message-body`
Maybe you want to simplify that to just point to one ID?

Reopening this for my feedback, however feel free to address it in debt week.
Thanks for improving this 👏

**Comment by sbatten** [comment]
> So there seems to be repetition "Do you trust the authors of the files in this folder" is announced twice. Could we improve that?

Based on the linked guidance I followed, the repetition is the price to pay to ensure the most accessible experience in all screen readers, I tried all sorts of tricks with focusing to get a single read through of the dialog on Voice Over, but I could not. Instead, I got this slightly repetitive but worked across screen readers option.

> Maybe you want to simplify that to just point to one ID?
The reason for this being multiple IDs is because the screen reader is smart enough to know how to read the contents in a visual way that makes the most sense. To simplify this from a code perspective, I would have to create my own transformation of the elements into a readable string (accounting for links and such within the text. I thought it was best to rely on the screen reader than ourselves.

Let me know what you think
